src/fft.jl

```diff
@@ -0,0 +1,576 @@
+using AbstractFFTs
+using LinearAlgebra
+using Dagger: DArray, @spawn, InOut, In
+using MPI
```

Comment:
```suggestion
```

We won't directly use MPI for the FFT implementation - Datadeps will handle this for us automatically, when necessary.